swarm/api/manifest_test.go

```diff
@@ -60,32 +61,53 @@ func checkEntry(t *testing.T, path, match string, trie *manifestTrie) {
 	} else if fullpath != match {
 		t.Errorf("incorrect entry retrieved for '%s'. expected path '%v', got '%s'", path, match, fullpath)
 	}
+
+	if multiple && entry.Status != http.StatusMultipleChoices {
```

Comment:
I think it would be good to test the else-condition:

```else if !multiple && entry.Status == http.StatusMultipleChoices```

in order to cover incorrect response status if we didn't expect multiple choices status, but got one.

Reply:
Added! Thanks for comment